### PR TITLE
[FLINK-12312][runtime] Remove CLI command for rescaling

### PR DESCRIPTION
**Pull request by GJL**
## What is the purpose of the change

*See http://apache-flink-user-mailing-list-archive.2336050.n4.nabble.com/DISCUSS-Temporarily-remove-support-for-job-rescaling-via-CLI-action-quot-modify-quot-td27447.html*

## Brief change log

  - *Remove rescaling code.*
  - *Throw exception from respective REST handlers.*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**yes** / no)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (**yes** / no / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/rescaling/RescalingHandlers.java [hunk 1]
```diff
@@ -108,21 +90,23 @@ public RescalingStatusHandler(
 		}
 
 		@Override
-		protected AsynchronousJobOperationKey getOperationKey(HandlerRequest<EmptyRequestBody, RescalingStatusMessageParameters> request) {
-			final JobID jobId = request.getPathParameter(JobIDPathParameter.class);
-			final TriggerId triggerId = request.getPathParameter(TriggerIdPathParameter.class);
+		public CompletableFuture<AsynchronousOperationResult<AsynchronousOperationInfo>> handleRequest(@Nonnull final HandlerRequest<EmptyRequestBody, RescalingStatusMessageParameters> request, @Nonnull final RestfulGateway gateway) throws RestHandlerException {
+			throw new RestHandlerException("Rescaling is temporarily disabled. See FLINK-12312.", HttpResponseStatus.SERVICE_UNAVAILABLE);
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/rescaling/RescalingHandlers.java]: Should be `throw featureDisabledException()`

Comment (GJL) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/rescaling/RescalingHandlers.java]: fixed

**Comment by GJL** [comment]
Thanks for the review. Merging as soon as travis gives green light.

**Comment by GJL** [comment]
Rebased to latest master.

**Pull request closed without merging**